apps/ui/src/stores/notifications.ts

```diff
@@ -0,0 +1,132 @@
+    {} as Record<string, number>
+  );
+
+  const route = useRoute();
+  const followedSpacesStore = useFollowedSpacesStore();
+  const metaStore = useMetaStore();
+  const { web3 } = useWeb3();
+
+  async function loadProposals(
+    network: Network,
+    current: number,
+    state: NonNullable<ProposalsFilter['state']>,
+    pivotTs: number,
+    spaceIds: string[]
+  ) {
+    return network.api.loadProposals(spaceIds, { limit: 100 }, current, {
+      state,
+      ...{ [state === 'closed' ? 'end_gte' : 'start_gte']: pivotTs }
+    });
+  }
+
+  async function loadNotifications() {
+    const now = Math.floor(Date.now() / 1e3);
+    const pivotTs = now - NOTIFICATION_TIME_WINDOW;
+
+    if (!followedSpacesStore.followedSpacesIds.length) return;
+
+    (Object.keys(followedSpacesStore.followedSpaceIdsByNetwork) as NetworkID[]).forEach(
+      async networkId => {
+        await metaStore.fetchBlock(networkId);
+      }
+    );
+
+    const promises = (
+      Object.entries(followedSpacesStore.followedSpaceIdsByNetwork) as [NetworkID, string[]][]
+    )
+      .map(([networkId, spaceIds]) => {
+        const network = getNetwork(networkId);
+        const current = metaStore.getCurrent(networkId) ?? 0;
+
+        return [
+          loadProposals(network, current, 'active', pivotTs, spaceIds),
+          loadProposals(network, current, 'closed', pivotTs, spaceIds)
+        ];
+      })
+      .flat();
+
+    const proposals = (await Promise.all(promises)).flat();
+
+    proposals.forEach(proposal => {
+      const timestamp = proposal.min_end < now ? proposal.min_end : proposal.start;
+      const lastUnread = lastUnreadTs.value[web3.value.account] ?? 0;
+
+      if (notifications.value.some(n => n.id === proposal.id)) return;
+
+      notifications.value.push({
+        id: proposal.id,
+        proposal,
+        type: proposal.min_end < now ? 'ended' : 'started',
+        timestamp,
+        unread: timestamp > lastUnread
+      });
+    });
+
+    notifications.value.sort((a, b) => b.timestamp - a.timestamp);
+  }
+
+  function markAllAsRead() {
+    notifications.value.forEach(notification => {
+      notification.unread = false;
+    });
+    lastUnreadTs.value[web3.value.account] = notifications.value[0]?.timestamp ?? 0;
+  }
+
+  const unreadNotificationsCount = computed(
+    () => notifications.value.map(n => n.unread).filter(Boolean).length
+  );
+
+  watch(
+    [() => followedSpacesStore.followedSpacesLoaded, () => followedSpacesStore.followedSpacesIds],
+    async loaded => {
+      if (!loaded) return;
+
+      loading.value = true;
+      notifications.value = [];
+      await loadNotifications();
+
+      loading.value = false;
+    },
+    { immediate: true }
+  );
+
+  onMounted(() => {
+    refreshNotificationInterval = window.setInterval(loadNotifications, REFRESH_INTERVAL * 1e3);
+    window.addEventListener(
+      'beforeunload',
+      () => String(route.name) === 'my-notifications' && markAllAsRead()
```

Comment:
Not ideal as we now have two places that handle it, especially given that it doesn't work in all cases still.

I was thinking:
1. To have separate ref for `unreadCount` in `Notifications` that tracks what counter should show (not particularly matching what current state is.
2. Use [`useWindowFocus`](https://vueuse.org/core/useWindowFocus/#usewindowfocus) and `onMounted` to  update `unreadCount` (to current `unreadNotificationsCount`) when mounting/focusing and **then** calling `markAllAsRead()` (this clears all notifications, but we still have reference what previous count was.
3. Watch `unreadNotificationsCount` and update `unreadCount`.

Reply:
I'll take a look at this approach

I tested, and `windowFocus` is not triggered when switching tab on mobile browser. Better to use https://vueuse.org/core/useDocumentVisibility/  I think

Reply:
But this approach of getting notifications count live is costly.

Since we show the counter in the Nav sidebar, means that we fetch the whole notifications content, on all page load, even if the user do not visit the notification page, since it's global

Reply:
> Since we show the counter in the Nav sidebar, means that we fetch the whole notifications content, on all page load, even if the user do not visit the notification page, since it's global

Isn't it the case already? I thought it's basically the same approach we have right now, but we would clear count before instead of after.

Reply:
Yes, my remark is completely unrelated to this particular code.

Reply:
Added a new `ref` to track the unread timestamp, so we can update the localstorage one when /notifications page is opened, instead of doing it only when closing.

Did not go the route of refreshing counter on page visibility/focus change